### PR TITLE
[2.14.x]DDF-04318 (with bug fix) Check that the user is logged in for every Intrigue WebSocket request

### PR DESCRIPTION
**Pull request by brjeter**
#### What does this PR do?
- **The first commit should go through an abbreviated review**; It's a clean cherry-pick of the commit from a PR that was already reviewed and merged (#4368) and then temporarily reverted (#4473) due to a bug.
- The second commit fixes the bug where certain error codes were not being handled on the frontend, causing Intrigue to not work properly in those error scenarios.

#### Who is reviewing it? 
@tyler30clemens @austinsteffes 
#### Ask 2 committers to review/merge the PR and tag them here.
@andrewkfiedler 
@tbatie

#### How should this be tested?
- Build DDF (include ui in the build)
- Install DDF

Verify the first commit's functionality works:
- Ingest some records and run some queries to make sure Intrigue is working normally.
- With one of the executed queries still up, open a new tab for the AdminUI and logout.
- Go back to Intrigue and edit the open query and/or execute it.
- The query should now return no results with an error icon next to the query.

Verify the second commit's bug fix:
- Create two Opensearch sources, 1 with a madeup hostname and 1 with the default loopback value.
- Try and execute the default wildcard search
- It should return records but show a red triangle indicating one of the sources is unavailable
#### What are the relevant tickets?
For GH Issues:
Fixes: #4318 

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: ui/packages/catalog-ui-search/src/main/webapp/js/model/QueryResponse.js [hunk 1]
```diff
@@ -139,13 +139,22 @@ module.exports = Backbone.AssociatedModel.extend({
             handled = true
             res.options = options
             switch (res.code) {
+              case 400:
+              case 404:
+              case 500:
```
Comment (brjeter) [inline ui/packages/catalog-ui-search/src/main/webapp/js/model/QueryResponse.js]: These status codes come from https://github.com/codice/ddf/blob/ddf-2.14.0/catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/QueryApplication.java#L193-L202

Comment (brjeter) [inline ui/packages/catalog-ui-search/src/main/webapp/js/model/QueryResponse.js]: These status codes come from https://github.com/codice/ddf/blob/ddf-2.14.0/catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/QueryApplication.java#L193-L202

**Comment by brjeter** [comment]
build now

**Comment by Kjames5269** [comment]
Hero starting...

**Comment by brjeter** [comment]
Closing in favor of the master port: https://github.com/codice/ddf/pull/4764

**Pull request closed without merging**